lms/services/segment.py

```diff
@@ -0,0 +1,86 @@
+from typing import TypedDict
+
+from sqlalchemy import func
+
+from lms.models import (
+    Grouping,
+    LMSCourse,
+    LMSSegment,
+    LMSSegmentMembership,
+    LMSUser,
+    LTIRole,
+)
+from lms.services.group_set import GroupSetService
+from lms.services.upsert import bulk_upsert
+
+
+class SegmentService:
+    def __init__(self, db, group_set_service: GroupSetService):
+        self._db = db
+        self._group_set_service = group_set_service
+
+    def upsert_segments(
+        self,
+        course: LMSCourse,
+        type_: Grouping.Type,
```

Comment:
Can you remind me what is the meaning of the underscore at the end of `type_`?

Reply:
`type` [is a built in function](https://docs.python.org/3/library/functions.html#type) just naming this `type` will mask that.

Instead of coming up with a new word for type we use the `_` to avoid the collision.